SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -256,6 +251,71 @@ bool USpatialSender::ValidateOrExit_IsSupportedClass(const FString& PathName)
 	return ClassInfoManager->ValidateOrExit_IsSupportedClass(RemappedPathName);
 }
 
+void USpatialSender::EntityComponentsDeleter::operator()(TArray<FWorkerComponentData>* Components) const noexcept
+{
+	if (Components == nullptr)
+	{
+		return;
+	}
+
+	for (FWorkerComponentData Component : *Components)
+	{
+		Schema_DestroyComponentData(Component.schema_type);
+	}
+
+	delete Components;
+}
+
+TArray<FWorkerComponentData> USpatialSender::CopyEntityComponentData(const EntityComponents& EntityComponents)
+{
+	TArray<FWorkerComponentData> Copy;
+	Copy.Reserve(EntityComponents->Num());
+	for (const FWorkerComponentData& Component : *EntityComponents)
+	{
+		Copy.Emplace(Worker_ComponentData{
+			Component.reserved,
+			Component.component_id,
+			Schema_CopyComponentData(Component.schema_type),
+			nullptr 
+		});
+	}
+
+	return Copy;
+}
+
+void USpatialSender::CreateEntityWithRetries(Worker_EntityId EntityId, FString EntityName)
+{
+	const EntityComponents& Components = EntitiesBeingCreatedWithRetries[EntityId];
+	const Worker_RequestId RequestId = Connection->SendCreateEntityRequest(CopyEntityComponentData(Components), &EntityId);
+
+	CreateEntityDelegate Delegate;
+
+	Delegate.BindLambda([this, EntityId, Name = MoveTemp(EntityName)](const Worker_CreateEntityResponseOp& Op) mutable
+	{
+		const EntityComponents& Components = EntitiesBeingCreatedWithRetries[EntityId];
```

Comment:
Doesn't look like this is needed

Reply:
It's used on the line below. Unless you're saying you think it would be easier to read inline. In which case I can change it.

Reply:
Read the wrong line. This isn't needed.